cermine-impl/src/main/java/pl/edu/icm/cermine/ContentExtractor.java

```diff
@@ -726,79 +738,64 @@ private Timeout combineWithMainTimeout(long timeoutSeconds) {
-                    + "                         time for a PDF file in seconds; by default, no\n"
-                    + "                         timeout is used; the value is approximate because in\n"
-                    + "                         some cases, the program might be allowed to slightly\n"
-                    + "                         exceeded this time, say by a second or two\n"
-                    + "  -configuration <path>	(optional) path to configuration properties file\n"
-                    + "                         see https://github.com/CeON/CERMINE\n"
-                    + "                         for description of available configuration properties\n"
-                    );
-            System.exit(1);
-        }
 
-        boolean override = parser.override();
-        Long timeoutSeconds = parser.getTimeout();
-        
-        String path = parser.getPath();
-        Map<String, String> extensions = parser.getTypesAndExtensions();
+    synchronized void printProgress(int total) {
+        progressCount++;
+        int percentage = progressCount * 100 / total;
+        System.out.println("Progress: " + percentage + "% done (" + progressCount + " out of " + total + ")");
+        System.out.println("");
+    }
 
-        File file = new File(path);
-        Collection<File> files = FileUtils.listFiles(file, new String[]{"pdf"}, true);
+    private static class parallelTask implements Runnable {
+        File file;
+        Map<String, String> extensions;
+        boolean override;
+        Long timeoutSeconds;
+        String outpath;
+        Long total;
 
-        ExtractionConfigBuilder builder = new ExtractionConfigBuilder();
-        if (parser.getConfigurationPath() != null) {
-            builder.addConfiguration(parser.getConfigurationPath());
+         parallelTask ( File files, Map<String, String> extensions, boolean override, Long timeoutSeconds, String outpath, Long limit) { 
+            this.file = files;
+            this.extensions = extensions;
+            this.override = override;
+            this.timeoutSeconds = timeoutSeconds;
+            this.outpath = outpath;
+            this.total = limit;
+        }
+
+        public void run() {
+            try {
+                runParallel();
+            }catch(IOException e) {
+                printException(e);
+            }catch(ParseException e) {
+                 printException(e);
+            }catch(AnalysisException e) {
+                printException(e);
+            }catch(TransformationException e) {
+                printException(e);
+            } 
+            
         }
-        builder.setProperty(ExtractionConfigProperty.IMAGES_EXTRACTION, extensions.containsKey("images"));
-        ExtractionConfigRegister.set(builder.buildConfiguration());
 
-        int i = 0;
-        for (File pdf : files) {
+        public void runParallel () throws ParseException, AnalysisException, IOException, TransformationException {
```

Comment:
I think there is no need for two separate methods in this class. The body of this method should be directly in `run()`.